packages/unified-selection/src/test/HiliteSetProvider.test.ts

```diff
@@ -332,6 +339,54 @@ describe("HiliteSetProvider", () => {
           sinon.match.any,
         );
       });
+
+      it("doesn't output duplicate instance IDs", async () => {
+        const elementKey1 = createSelectableInstanceKey(1, "TestSchema:TestClass");
+        const customElement1 = createCustomSelectable(1, [{ className: "TestSchema:TestClass", id: "0x1" }]);
+
+        mockClass("TestClass", "BisCore", "Element");
+        mockQuery([], [], ["0x1"]);
+
+        const selection = Selectables.create([elementKey1, customElement1]);
+        const result = await loadHiliteSet(selection);
+        expect(result.elements).to.deep.eq(["0x1"]);
+      });
+
+      it("returns values in time intervals", async () => {
+        const timers = sinon.useFakeTimers();
+        const elementPromises = [new ResolvablePromise<string>(), new ResolvablePromise<string>()];
+        const elementKeys = [1, 2].map((i) => createSelectableInstanceKey(i, "TestSchema:TestElement"));
+
+        mockClass("TestElement", "BisCore", "Element");
+        mockQuery([], [], elementPromises);
+
+        const selectables = Selectables.create(elementKeys);
+        const iter = provider.getHiliteSet({ selectables });
+        elementPromises[0].resolve(elementKeys[0].id);
+
+        const nextValuePromise = iter.next();
+        await Promise.all([expect(waitForPromise(nextValuePromise, 5)).to.eventually.be.undefined, timers.tickAsync(5)]);
```

Comment:
I'm not a fan of the `waitForPromise` function... Maybe just do this?
```
expect(await Promise.race([
  nextValuePromise,
  timers.tickAsync(5).then(() => "timeout"),
])).to.eq("timeout");
```

Reply:
Agree, applied suggested changes